modules/write_gtc.py

```diff
@@ -0,0 +1,121 @@
+from IlluminaBeadArrayFiles import GenotypeCalls, BeadArrayUtility
+import struct
+from io import BytesIO
+import logging
+
+logger = logging.getLogger('write_gtc')
+logger.debug('In module write_gtc.py')
+
+def handle_int(value):
+    return struct.pack("<i", value)
+
+def handle_short(value):
+    return struct.pack("<H", value)
+
+def handle_char(value):
+    return struct.pack("c", value)
+
+def handle_byte(value):
+    #print(value)
+    return struct.pack("B", value)
+
+def handle_float(value):
+    return struct.pack("<f", value)
+
+def handle_gc50(value):
+    return struct.pack("<fiii", value[0], value[1], value[2], value[3])
+
+def handle_percentiles(value):
+    return struct.pack("<HHH", value[0], value[1], value[2])
+
+def handle_string(value):
+    assert len(value) <= 127
+    return struct.pack("B", len(value)) + value
+
+def handle_basecalls(value):
+    return value
+
+def handle_scanner_data(value):
+    return handle_string(value.name) + handle_int(value.pmt_green) + handle_int(value.pmt_red) + handle_string(value.version) + handle_string(value.user)
+
+def handle_normalization_transform(value):
+#    return struct.pack("<iffffff", value.version, value.offset_x, value.offset_y, value.scale_x, value.scale_y, value.shear, value.theta)
+    return struct.pack("<iffffffiiiiii", value.version, value.offset_x, value.offset_y, value.scale_x, value.scale_y, value.shear,

```

Comment:
This is the byte shift -- prior it was ""<iffffff" and now it has been updated to "<iffffffiiiiii"

Reply:
Also, adding a comma-separate list of 0 is a new addition to the struct.pack()